src/ucp/proto/proto_common.c

```diff
@@ -521,45 +622,68 @@ ucp_proto_common_find_am_bcopy_hdr_lane(const ucp_proto_init_params_t *params)
     return lane;
 }
 
-ucs_linear_func_t
-ucp_proto_common_memreg_time(const ucp_proto_common_init_params_t *params,
-                             ucp_md_map_t reg_md_map)
+void ucp_proto_common_memreg_time(const ucp_proto_common_init_params_t *params,
+                                  ucp_md_map_t reg_md_map,
+                                  ucs_linear_func_t *memreg_time,
+                                  ucp_proto_perf_node_t **perf_node_p)
 {
-    ucp_context_h context      = params->super.worker->context;
-    ucs_linear_func_t reg_cost = UCS_LINEAR_FUNC_ZERO;
+    ucp_context_h context = params->super.worker->context;
+    ucp_proto_perf_node_t *perf_node;
     const uct_md_attr_t *md_attr;
     ucp_md_index_t md_index;
+    const char *md_name;
+
+    *memreg_time = UCS_LINEAR_FUNC_ZERO;
+
+    if (reg_md_map == 0) {
+        *perf_node_p = NULL;
+        return;
+    }
+
+    perf_node = ucp_proto_perf_node_new_data("mem reg", "");
 
     /* Go over all memory domains */
     ucs_for_each_bit(md_index, reg_md_map) {
         md_attr = &context->tl_mds[md_index].attr;
-        ucs_linear_func_add_inplace(&reg_cost, md_attr->reg_cost);
-        ucs_trace("md %s" UCP_PROTO_PERF_FUNC_FMT(reg_cost),
-                  context->tl_mds[md_index].rsc.md_name,
+        md_name = context->tl_mds[md_index].rsc.md_name;
+        ucs_linear_func_add_inplace(memreg_time, md_attr->reg_cost);
+        ucs_trace("md %s" UCP_PROTO_PERF_FUNC_FMT(reg_cost), md_name,
                   UCP_PROTO_PERF_FUNC_ARG(&md_attr->reg_cost));
+
+        ucp_proto_perf_node_add_data(perf_node, md_name, md_attr->reg_cost);
     }
 
-    return reg_cost;
+    if (ucs_popcount(reg_md_map) > 1) {
```

Comment:
minor: maybe use `is_power_of2` instead?

Reply:
do you think it's not worth it, because it is not a fast path?

Reply:
I think I fixed it:
https://github.com/openucx/ucx/pull/8282/commits/ea3c84fa2960fa9cf0a0ab53eb45ee9e484f303b#diff-9402784f6eb886015832f7c7ffd1ed06df79272660219f51ef42c6b7eac39ef3R648

Reply:
oh it's another place